### ISSUE TITLE
Ensemble methods

### ISSUE THREAD

**Issue by johannbrehmer** [johannbrehmer/madminer#104]
We could / should implement a new class that simplified ensemble methods for likelihood ratio / Fisher information estimation. This would give out more precise predictions plus an error estimate.

Instead of a democratic vote, we could use the knowledge that ideally E[r] = 0 and E[t] = 0 (with the correct sampling), and use a weighted meta-prediction like
```
r_hat(x)  =  (sum_i r_hat_i(x) exp(-beta E[r_hat_i(x)])) / (sum_i exp(-beta E[r_hat_i(x)]))
```
where `i` are the different estimators.

I'm not sure whether the ensemble variance is still an estimator of the estimator variance if we use weights like this.

**Comment by johannbrehmer** [comment]
Not sure if we'd have to reserve a separate validation sample for calculating these expectations, or if it's fine to re-use the training data since the E[r] / E[t] property was never used in the losses.

**Comment by johannbrehmer** [comment]
Implemented in 4e04a6259757a97bf72b41df1baa39b2173646e8
